src/extension.ts

```diff
@@ -265,9 +265,12 @@ class MarkdownTocTools {
 
     private createToc(editBuilder : TextEditorEdit, headerList : any[], insertPosition : Position) {
         let lineEnding      = <string>  workspace.getConfiguration("files").get("eol");
+        if (lineEnding === "auto") {
+            lineEnding = "\n";
```

Comment:
I think it should come from the current document 🤔

Reply:
Right, should be better taken either from current document, or probably vscode provides some API to resolve **auto** to os-specific value.